tests/deprecated_api/test_remove_1-4.py

```diff
@@ -189,3 +189,18 @@ def test_v1_4_0_deprecated_lightning_data_parallel():
         dp_model = LightningDataParallel(model, device_ids=[0])
     assert isinstance(dp_model, torch.nn.DataParallel)
     assert isinstance(dp_model.module, LightningParallelModule)
+
+
+def test_v1_4_0_deprecated_checkpoint_on(tmpdir):
+    from pytorch_lightning.callbacks.model_checkpoint import warning_cache
+    warning_cache.clear()
+
+    class TestModel(BoringModel):
+        def training_step(self, batch, batch_idx):
+            self.log("val_loss", -batch_idx)
+            return super().training_step(batch, batch_idx)
+
+    trainer = Trainer(default_root_dir=tmpdir, checkpoint_callback=True, max_epochs=1)
```

Comment:
```suggestion
    trainer = Trainer(default_root_dir=tmpdir, callbacks=[ModelCheckpoint()], max_epochs=1)
```
checkpoint_callback will be removed in 1.3. Also, you might need to import `ModelCheckpoint` here.

Reply:
`checkpoint_callback` won't be removed, what will be removed is passing an instance of `ModelCheckpoint` to `checkpoint_callback`

Reply:
correct :+1: